src/core/packages/saved-objects/migration-server-internal/src/zdt/utils/outdated_documents_query.ts

```diff
@@ -21,20 +21,22 @@ export const getOutdatedDocumentsQuery = ({
   // Note: in theory, we could check the difference of model version with the index's
   // and narrow the search filter only on the type that have different versions.
   // however, it feels safer to just search for all outdated document, just in case.
-  const virtualVersions = getVirtualVersionMap(types);
+  const virtualVersions = getVirtualVersionMap({ types, useModelVersionsOnly: false });
   return {
     bool: {
-      should: types.map((type) => {
-        const virtualVersion = virtualVersions[type.name];
-        return {
-          bool: {
-            must: [
-              { term: { type: type.name } },
-              { range: { typeMigrationVersion: { lt: virtualVersion } } },
-            ],
-          },
-        };
-      }),
+      should: types
+        .filter((type) => virtualVersions[type.name] !== '0.0.0') // types with no migrations or modelVersions cannot be outdated
```

Comment:
I added this filter, as documents with no modelVersions and no migrations will have a virtualVersion of `0.0.0`.
Nonetheless, this does not mean they are outdated, they simply never defined any transformation.

For the record, the logic used to return `0.0.0` before this PR as well.